lib/chain.js

```diff
@@ -0,0 +1,238 @@
+    timespan = this.targetTimespan * 4;
+  }
+
+  var oldTarget = this.getTargetFromBits(bits);
+  var newTarget = oldTarget.mul(new BN(timespan, 10)).div(new BN(this.targetTimespan, 10));
+  var newBits = this.getBitsFromTarget(newTarget);
+
+  if(newBits > this.maxBits) {
+    newBits = this.maxBits;
+  }
+
+  return newBits;
+};
+
+/**
+ * Calculates the number of blocks for a retargeting interval
+ * @param {Block} - block - An instance of a block
+ * @param {Function} - callback - A callback function that accepts arguments: Error and Number
+ */
+Chain.prototype.getNextWorkRequired = function getNextWorkRequired(block, callback) {
+
+  var self = this;
+
+  var interval = this.getDifficultyInterval();
+
+  self.getHeightForBlock(block.hash, function(err, height) {
+
+    if (err) {
+      return callback(err);
+    }
+
+    if (height === 0) {
+      return callback(null, self.maxBits);
+    }
+
+    // not on interval, return the same amount of difficulty
+    if ((height + 1) % interval !== 0) {
+      return callback(null, block.bits);
+    }
+
+    // otherwise compute the new difficulty
+    self.getBlockAtHeight(block, height + 1 - interval, function(err, lastIntervalBlock){
+      if (err) {
+        callback(err);
+      }
+
+      var timespan = (Math.floor(block.timestamp.getTime() / 1000) * 1000) - (Math.floor(lastIntervalBlock.timestamp.getTime() / 1000) * 1000);
+      var bits = self.getRetargetedBits(lastIntervalBlock.bits, timespan);
+
+      return callback(null, bits);
+
+    });
+
+  });
+
+};
+
+/**
+ * Calculates the actual target from the compact form
+ * @param {Number} - bits
+ * @returns {BN}
+ */
+Chain.prototype.getTargetFromBits = function getTargetFromBits(bits) {
+  if(bits <= this.minBits) {
+    throw new Error('bits is too small (' + bits + ')');
+  }
+
+  if(bits > this.maxBits) {
+    throw new Error('bits is too big (' + bits + ')');
+  }
+
+  var a = bits & 0xffffff;
+  var b = bits >>> 24;
+
+  var exp = (8 * (b - 3));
+
+  // Exponents via bit shift (works for powers of 2)
+  var z = (new BN(2, 10)).shln(exp - 1);
+  var target = (new BN(a, 10)).mul(z);
+
+  return target;
+
+};
+
+/**
+ * Calculates the compact target "bits" from the target
+ * @param {BN|Number} - target
+ * @returns {Number}
+ */
+Chain.prototype.getBitsFromTarget = function getBitsFromTarget(target) {
+  target = new BN(target, 'hex');
+
+  var tmp = target;
+
+  var b = 0;
+  while(tmp.cmp(new BN(0, 10)) > 0) {
+    b++;
+    tmp = tmp.shrn(8);
+  }
+
+  var a = target.shrn((b - 3) * 8);
+  var bits = Number('0x' + b.toString(16) + a.toString(16, 6));
+  return bits;
+};
+
+Chain.prototype.getDifficultyFromBits = function getDifficultyFromBits(bits) {
+  var currentTarget = this.getTargetFromBits(bits);
+  var genesisTarget = this.getTargetFromBits(this.genesis.bits);
+  return genesisTarget.div(currentTarget);
+};
+
+Chain.prototype.getBlockWeight = function getBlockWeight(blockHash, callback) {
```

Comment:
A method to get this directly from Bitcoin Core is ready in PR: https://github.com/bitpay/bitcoind.js/pull/32